sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/XXE/XmlReaderSettingsValidator.cs

```diff
@@ -28,39 +29,54 @@
 
 namespace SonarAnalyzer.Rules.XXE
 {
+    /// <summary>
+    /// This class is responsible to check if a XmlReaderSettings node is vulnerable to XXE attacks.
+    ///
+    /// By default the class is safe:
+    ///  - before .Net 4.5.2 it has XmlResolver set to an instance of XmlUrlResolver which is not secure but since has ProhibitDtd set to false it is safe to use
+    ///  - starting with .Net 4.5.2 XmlResolver is set to null, ProhibitDtd (now obsolete) set to false and DtdProcessing set to ignore
+    ///
+    /// If the properties are modified, in order to be secure, we have to check that either ProhibitDtd is set to false, DtdProcessing set to Ignore or
+    /// XmlResolver is set to null or is different than XmlUrlResolver and XmlPreloadedResolver.
```

Comment:
```suggestion
    /// the internal XmlResolver is secure.
```

Reply:
the actual checks may change in the future (maybe instead of using a blacklist we'll use a whitelist) and can be verified in the code itself.